python/pyarrow/includes/libarrow.pxd

```diff
@@ -1788,6 +1788,25 @@ cdef extern from "arrow/compute/api.h" namespace "arrow::compute" nogil:
         CPartitionNthOptions(int64_t pivot)
         int64_t pivot
 
+    enum CSortOrder" arrow::compute::SortOrder":
```

Comment:
I think this should be `ctypedef enum` because `arrow::compute::SortOrder` is a scoped num (it's declared as `enum class SortOrder`.

Hopefully that'd fix the compile error on CI.

Reply:
Ah, I only looked at the 32bit failure, which seemed unrelated. Will push a fix.